### PR TITLE
WS2-1623: Create MVP version of monorepo for ASU modules

### PR DESCRIPTION
**Pull request by tbutterf**
### Description

As per our conversation a few days ago, I have gone with an MVP approach. In order to make it a truly viable product, I needed to do quite a bit of dependency cleanup. So, it turns out that in order to have the asu_brand module, we need 4 other modules. So, I have added all five to the MVP effort and am now ready for PR review. The included modules are:

- asu_brand
- asu_react_core
- asu_react_integration
- asu_config_utility (I split out this functionality from the webspark_utility module so that we have a tool just for config management)
- asu_user

QA Steps:
The Github Actions integration has been tested and has passed QA. But, to test whether the asu_brand module is able to be added to a non-Webspark site, you can do the following:
- Follow these directions to spin up a vanilla Drupal 10 site locally in DDEV: https://ddev.readthedocs.io/en/stable/users/quickstart/#drupal-drupal-10
- Edit the composer.json file to add "asuwebplatforms/asu_brand": "2.14.1" to the “require” section.
- Edit the "minimum-stability" key to be "dev" instead of "stable" and save the file.
- Run ddev composer update
- Launch the site and after initializing it, attempt to enable the asu_brand module by navigating to /admin/modules and clicking the checkbox next to the module, and then clicking “Install.” You should see a message that says You must install the ASU React Integration, ASU React Core, Paragraphs, Entity Reference Revisions, ASU User, CAS, External Authentication, ASU Config Utility, Configuration Update Base modules to install ASU Brand. Click "Continue."
- Navigate to /admin/structure/block and click the “Place block” button in the Header region.
- Click the “Place block” button on the row for “ASU brand header” and click “Save block”
- Navigate to the home page. Now you should see the ASU header on the page. If so, this passes.

### Links

- [JIRA ticket](https://asudev.jira.com/browse/WS2-1623)

### Checklist

- [x] Design updates match [Web Standards](https://xd.adobe.com/view/56f6cb78-9af5-4b12-b4ce-ef319f71113f-03a5/) and [Unity Design System](https://unity.web.asu.edu)
- [x] Solution is documented on the Jira ticket
- [x] QA steps to verify have been included on the Jira ticket
- [x] No new PHP or JS errors
- [x] No accessibility issues are introduced with this update
- [x] Added/updated README.md files, if relevant
- [x] Confirm that any yaml files included have a matching update hook

### Verified in browsers 

- [x] Chrome
- [x] Safari
- [x] Firefox
- [x] Edge

### Screenshots

Note: Sections that are not applicable for this issue can be removed.

### CODE REVIEW

**Comment by mlsamuelson** [comment]
[Visit Site](https://pr-613-webspark-ci.pantheonsite.io/)

Created multidev environment [pr-613](https://dashboard.pantheon.io/sites/aec2f75f-35eb-41f8-abe7-95b292415259#pr-613) for webspark-ci.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
@tbutterf Looks like in the moving of the modules the .gitignore files in them (which are key for ensuring the right package files get included for components) were not moved and are being removed. Can you make sure they make the move please?

**Comment by tbutterf** [comment]
@mlsamuelson I have fixed the .gitignore issue.

File: web/modules/webspark/asu_degree_rfi/config/install/field.field.node.degree_listing_page.field_degree_detail_breadcrumbs.yml
Comment (mlsamuelson) [inline web/modules/webspark/asu_degree_rfi/config/install/field.field.node.degree_listing_page.field_degree_detail_breadcrumbs.yml:1]: Was this an accidental removal?

File: web/profiles/webspark/webspark/webspark.info.yml
Comment (mlsamuelson) [inline web/profiles/webspark/webspark/webspark.info.yml:1]: Long term it's good to get these sorted in alpha. And the number of lines is increased by one, I presume for the addition of asu_config_utility. I spot checked changed lines to verify nothing went missing.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]

File: web/profiles/webspark/webspark/webspark.install
Comment (mlsamuelson) [inline web/profiles/webspark/webspark/webspark.install:745]: Update hook here to enable the new module is great.

The old historical updates are going to be problematic for older sites updating that haven't hit this yet. Two possible solves: 
1. add a helper function to the profile that can be used to check if the new module is enabled and does so if it isn't and then add that to the start of all the historical updates. OR
2. keep the historical config update service in place and don't update the past update hooks, and we'll have some duplicated and deprecated code for a bit, but we can remove it eventually. All new code can use the new asu config utility moving forward.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]

File: web/profiles/webspark/webspark/webspark.install
Comment (tbutterf) [reply web/profiles/webspark/webspark/webspark.install:745]: @mlsamuelson I have decided to go with option #2.

File: web/modules/webspark/asu_degree_rfi/config/install/field.field.node.degree_listing_page.field_degree_detail_breadcrumbs.yml
Comment (tbutterf) [reply web/modules/webspark/asu_degree_rfi/config/install/field.field.node.degree_listing_page.field_degree_detail_breadcrumbs.yml:1]: Yes. Not sure how that happened.

Comment (tbutterf) [reply web/modules/webspark/asu_degree_rfi/config/install/field.field.node.degree_listing_page.field_degree_detail_breadcrumbs.yml:1]: After looking into this more, I have concluded that this actually was intended. There were two copies of this file, so this deletes the duplicate. The other one was moved into the new asu_modules directory.

**Comment by tbutterf** [comment]
@mlsamuelson and @bebaps This is ready now.

**Review by bebaps — COMMENTED** [review]
Followed the QA steps and it does work. I encountered two things of note:

1. On the vanilla site, when you go to edit the Main Navigation, I am getting an error from the `asu_brand` module (see screenshot). 
2. On the vanilla site, on the main navigation edit screen there is no text stating "First menu link automatically becomes the home link"

**Comment by mlsamuelson** [comment]
@bebaps Did you forget to add the screenshot? Not needed, but what's the error say?

**Comment by bebaps** [comment]
@mlsamuelson yea I forgot to add it, I had to start another task and came back to this. Screenshot added.

Screenshot 2024-11-07 at 3 47 13 PM

**Comment by mlsamuelson** [comment]
Perhaps outside of the WS2 context, there's some default configs that aren't in place... 

The PR already touches the `asu_brand` code so I'll leave this as your call, @tbutterf if you think this is in-bounds for this PR, or if it should be a separate follow-up ticket.

**Comment by tbutterf** [comment]
I have responded to @bebaps's concerns in Slack: https://asu-tech.slack.com/archives/CL512LPAT/p1731083971644029. I'm not sure which default configs aren't in place, @mlsamuelson. But, if you're referring to Alexander's concerns, I think I have resolved those concerns. Was there something else?

**Comment by mlsamuelson** [comment]
@tbutterf @bebaps PR has conflicts now. Looks like an easy one to fix on the command line... Travis, I think this is good to go (apart from the conflicts), after the last conversations we had about ASU Brand yesterday... Correct?

**Review by mlsamuelson — APPROVED** [review]
Apart from the conflicts to resolve, this looks good. Approving.

**Pull request merged**
